Documentation/GettingStarted.md

```diff
@@ -1,104 +1,861 @@
+        return create { observer in
+            let task = self.dataTaskWithRequest(request) { (data, response, error) in
+                if data == nil || response == nil {
+                    sendError(observer, error ?? UnknownError)
+                }
+                else {
+                    sendNext(observer, (data, response))
+                    sendCompleted(observer)
+                }
+            }
+
+            task.resume()
+
+            return AnonymousDisposable {
+                task.cancel()
+            }
+        }
+    }
+}
+```
+
+## Operators
+
+There are numerous operators implemented in RxSwift. Complete list can be found [here](API.md).
+
+Marble diagrams for all operators can be found on [ReactiveX.io](http://reactivex.io/)
+
+Almost all operators are demonstrated in [Playgrounds](../Playgrounds).
+
+To use playgrounds please open `Rx.xcworkspace`, build `RxSwift-OSX` scheme and then open playgrounds in `Rx.xcworkspace` tree view.
+
+In case you need an operator, and don't know how to find it there a [decision tree of operators]() http://reactivex.io/documentation/operators.html#tree).
+
+[Supported RxSwift operators](API.md#rxswift-supported-operators) are also grouped by function they perform, so that can also help.
+
+### Custom operators
+
+There are two ways how you can create custom operators.
+
+#### Easy way
+
+All of the internal code uses highly optimized versions of operators, so they aren't the best tutorial material. That's why it's highly encouraged to use standard operators.
+
+Fortunately there is an easier way to create operators. Creating new operators is actually all about creating observables, and previous chapter already describes how to do that.
+
+Lets see how an unoptimized map operator can be implemented.
+
+```
+func myMap<E, R>(transform: E -> R)(source: Observable<E>) -> Observable<R> {
+    return create { observer in
+
+        let subscription = source >- subscribe { e in
+                switch e {
+                case .Next(let boxedValue):
+                    sendNext(observer, transform(boxedValue.value))
+                case .Error(let error):
+                    sendError(observer, error)
+                case .Completed:
+                    sendCompleted(observer)
+                }
+            }
+
+        return subscription
+    }
+}
+```
+
+So now you can use your own map:
+
+```
+let subscription = myInterval(0.1)
+    >- myMap { e in
+        return "This is simply \(e)"
+    }
+    >- subscribeNext { n in
+        println(n)
+    }
+```
+
+and this will print
+
+```
+Subscribed
+This is simply 0
+This is simply 1
+This is simply 2
+This is simply 3
+This is simply 4
+This is simply 5
+This is simply 6
+This is simply 7
+This is simply 8
+...
+```
+
+#### Harder, more performant way
+
+You can perform the same optimizations like we have made and create more performant operators. That usually isn't necessary, but it ofc can be done.
```

Comment:
> ofc

This feels weird in here, happy with it either fully written out as "of course", or omitted.